include/circt/Dialect/Moore/MooreOps.td

```diff
@@ -970,11 +972,11 @@ def StructCreateOp : MooreOp<"struct_create", [SameOperandsAndResultType]> {
     ```
     See IEEE 1800-2017 § 7.2. "Structures".
   }];
-  let arguments = (ins UnpackedType:$input);
-  let results = (outs UnpackedType:$result);
-  let hasCustomAssemblyFormat = 1;
+  let arguments = (ins StrAttr:$name,  Variadic<UnpackedType>:$input);
+  let results = (outs AnyStructType:$result);
   let assemblyFormat = [{
-  $input attr-dict `:` type($input)
+    `` custom<ImplicitSSAName>($name) $input attr-dict
+    `:` type($input) `->` type($result)
   }];
```

Comment:
Can you add a constraint like `TypesMatchWith<...>` or a custom verifier that checks if the types of the `input`s are exactly equal to the types of the result struct fields? Otherwise this would allow you to do something like `moore.struct_create %a, %b : i42, string -> struct<{a: event, b: l9001}>`.

Reply:
It seems TypeMatchWith<> can not fit this case, I just wrote a verifier for union and struct. Please have a look, thanks a lot.